cluster/eks/composition.yaml

```diff
@@ -37,18 +37,18 @@ spec:
               }
       name: controlplaneRole
     - base:
-        apiVersion: iam.aws.crossplane.io/v1beta1
+        apiVersion: iam.aws.upbound.io/v1beta1
         kind: RolePolicyAttachment
         spec:
           forProvider:
             policyArn: arn:aws:iam::aws:policy/AmazonEKSClusterPolicy
-            roleNameSelector:
+            roleSelector:
               matchControllerRef: true
               matchLabels:
                 role: controlplane
       name: clusterRolePolicyAttachment
     - base:
-        apiVersion: eks.aws.crossplane.io/v1beta1
+        apiVersion: eks.aws.upbound.io/v1beta1
```

Comment:
ClusterAuth is ready to be used to get the kubeconfig https://marketplace.upbound.io/providers/upbound/provider-aws/v0.14.0/resources/eks.aws.upbound.io/ClusterAuth/v1beta1